compiler+runtime/src/cpp/jank/codegen/llvm_processor.cpp

```diff
@@ -298,14 +306,35 @@ namespace jank::codegen
   /* NOLINTNEXTLINE(readability-make-member-function-const): Can't be const, due to overload resolution ambiguities. */
   llvm::Value *llvm_processor::gen(expr::var_deref_ref const expr, expr::function_arity const &)
   {
-    auto const ref(gen_var(make_box<obj::symbol>(expr->var->n, expr->var->name)));
-    auto const fn_type(
-      llvm::FunctionType::get(ctx->builder->getPtrTy(), { ctx->builder->getPtrTy() }, false));
-    auto const fn(ctx->module->getOrInsertFunction("jank_deref", fn_type));
-
-    llvm::SmallVector<llvm::Value *, 1> const args{ ref };
-    auto const call(ctx->builder->CreateCall(fn, args));
+    llvm::Value *call{};
+    auto const var_qualified_name(make_box<obj::symbol>(expr->var->n, expr->var->name));
+    /*
+     * For direct-calls, when derefing a var we need to handle two different types of var-derefs.
+     * First, we only direct-call vars that are not dynamic.
+     * When generating the IR for a var-root, if the function is a jank_load function, the var_root is derefed directly in the jank_load functoin.
```

Comment:
1. Spelling error: functoin
2. Please follow the style guidelines with regard to line length. We have a column limit of 100.
3. We say say there are two things and we identify"first", but not not "second"